src/backend/distributed/utils/replication_origin_session_utils.c

```diff
@@ -0,0 +1,263 @@
+/*-------------------------------------------------------------------------
+ *
+ * replication_origin_session_utils.c
+ *   Functions for managing replication origin session.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "distributed/replication_origin_session_utils.h"
+#include "distributed/remote_commands.h"
+#include "distributed/metadata_cache.h"
+#include "utils/builtins.h"
+#include "miscadmin.h"
+
+static bool IsRemoteReplicationOriginSessionSetup(MultiConnection *connection);
+
+static bool ExecuteRemoteCommandAndCheckResult(MultiConnection *connection, char *command,
+											   char *expected);
+
+static inline bool IsLocalReplicationOriginSessionActive(void);
+
+PG_FUNCTION_INFO_V1(replication_origin_session_start_no_publish);
+PG_FUNCTION_INFO_V1(replication_origin_session_end_no_publish);
+PG_FUNCTION_INFO_V1(replication_origin_session_is_no_publish);
+
+/*
+ * This variable is used to remember the replication origin id of the current session
+ * before resetting it to DoNotReplicateId in SetupReplicationOriginLocalSession.
+ */
+static RepOriginId originalOriginId = InvalidRepOriginId;
```

Comment:
minor: we usually start global variables with a capital letter

Reply:
OK. renamed it to OriginalOriginId